middleware.ts

```diff
@@ -54,7 +54,10 @@ const loadSession = async (request: NextRequest, sessionToken?: string) => {
 export async function middleware(req: NextRequest) {
   const res = NextResponse.next();
 
-  if (!pathsToMatch.some((matcher) => pathToRegexp(matcher).test(req.nextUrl.pathname))) {
+  if (
```

Comment:
There's still the Next.js middleware issue mentioned in the comment in middleware code, https://github.com/open-sauced/insights/pull/1549, so just adding another check here so that `redirectFrom` is handled by middleware if you're logged in.